### ISSUE TITLE
Checkpoint 2 ready for review

### ISSUE THREAD

**Issue by Carson-Ray** [Carson-Ray/indieProject#5]
@pawaitemadisoncollege checkpoint 2 is ready for review

**Comment by pawaitemadisoncollege** [comment]
Looks like some great progress here so far @Carson-Ray!

- [x] Is there an ERD here? If not, it can be generated pretty easily in IntelliJ: https://www.jetbrains.com/help/idea/creating-diagrams.html. Seeing a visual will help me provide better feedback on that aspect of the design.
- [x] This would be a good time to re-evaluate the project plan, moving things forward if they were not completed in the originally planned timeframe.
- [x] Think about testing the delete scenarios in one-to-many relationships more fully. For example, if a user is deleted, what should happen to the associated Pokeman? What if a Pokemon is deleted? Write tests to make sure whatever should happen, does happen.
- [x] You've probably noticed that writing a Dao for each entity can be very repetitive and produce a lot of duplicate good - this is a good time swap in the GenericDao and remove the current entity-specific daos. Note that you will still have unit tests  for each entity(dao) to make sure that annotations and database relationships are set up correctly.

Looking forward to seeing data on the front end of the next checkpoint!

**Comment by Carson-Ray** [comment]
Hi @pawaitemadisoncollege , I made the changes that you pointed out. I created an ERD that is now pushed, evaluated where I am at with my plan (as discussed in class today), updated my delete scenarios to ensure that deleting a user deletes the associated pokemon as well (and that deleting a pokemon does NOT delete the associated user, and implemented the generic dao, though I am not totally certain that I did so correctly. Apologies for this very long run-on sentence, and for the amount of time it has taken.

**Comment by pawaitemadisoncollege** [comment]
Hi @Carson-Ray! Glad to hear that things are rolling along!

- [x] Can you point me to the ERD? I didn't see it at first glance through your repository (providing it as an image file is best for easy viewing in GitHub).
- [x] I think your implementation of the GenericDao looks good. To truly test it to confirm complete implementation and functionality, remove the PokemonDao and UserDao classes and verify the unit tests all still pass.

**Comment by Carson-Ray** [comment]
Done and done. The other Dao classes are gone and an actual image file is there for the ERD

**Comment by pawaitemadisoncollege** [comment]
Nice work on the Dao conversion!

One quick thing about the User table, will you need password on that table if using cognito?

**Comment by Carson-Ray** [comment]
You are absolutely right, I had that in the table prior to knowing much about Cognito. Will be removing that today. Thank you!
